tools/validators/instance_validator/parser.py

```diff
@@ -0,0 +1,56 @@
+from strictyaml import Map, MapPattern, Str, Optional, Any, load, Enum, Regex, Seq, YAMLValidationError
+
+# TODO check all valid states and ontological references in next validation steps
+schema = MapPattern(Str(), 
+    Map({
+        'type': Str(), 
+        'id': Str(), 
+        Optional('connections'): MapPattern(Str(), Str())
+                               | Seq(MapPattern(Str(), Str())),
+        Optional('links'): MapPattern(Str(), 
+            MapPattern(Str(), Str())),
+        Optional('translation'): Any(),
+        Optional('metadata'): Any()
+    }))
+
+translation_schema = Str() | Any()
+
+# TODO add manual check for translation_data_schema to de-duplicate units/unit_values/states
+# TODO add all units/unit_values/states to translation_data_schema
+translation_data_schema = Str() | Map({
+                                    'present_value': Str(),
+                                    Optional('states'): MapPattern(Str(), Str()),
+                                    Optional('units'): Map({
+                                        'key': Str(),
+                                        'values': MapPattern(Str(), Str())
+                                    }),
+                                    Optional('unit_values'): MapPattern(Str(), Str())
+                                })
+
+def load_yaml_with_schema(filepath, schema):
```

Comment:
please add documentation to your functions

Reply:
Done